File: code_challenge_solution/javascript/20 Calculate Dot and Cross Product of Two Vectors.js
const vector1 = [2, 4, 6];
const vector2 = [1, 3, 5];

// Calculate dot product
let dotProduct = 0;
for (let i = 0; i < vector1.length; i++) {
    dotProduct += vector1[i] * vector2[i];
}

// Calculate cross product
const crossProduct = [
    vector1[1] * vector2[2] - vector1[2] * vector2[1],
    vector1[2] * vector2[0] - vector1[0] * vector2[2],
    vector1[0] * vector2[1] - vector1[1] * vector2[0]
];

console.log("Dot Product: " + dotProduct);
console.log("Cross Product: " + JSON.stringify(crossProduct));

File: code_challenge_solution/c++/20 Calculate Dot and Cross Product of Two Vectors.cpp
#include<iostream>
#include<vector>

int dotProduct(std::vector<int> v1, std::vector<int> v2) {
    int dotProduct = 0;
    for (int i = 0; i < v1.size(); i++) {
        dotProduct += v1[i] * v2[i];
    }
    return dotProduct;
}

std::vector<int> crossProduct(std::vector<int> v1, std::vector<int> v2) {
    std::vector<int> crossProduct(3);
    crossProduct[0] = v1[1] * v2[2] - v1[2] * v2[1];
    crossProduct[1] = v1[2] * v2[0] - v1[0] * v2[2];
    crossProduct[2] = v1[0] * v2[1] - v1[1] * v2[0];
    return crossProduct;
}

int main() {
    std::vector<int> v1 = {1, 2, 3};
    std::vector<int> v2 = {4, 5, 6};

    std::cout << "Dot Product: " << dotProduct(v1, v2) << std::endl;

    std::vector<int> cross = crossProduct(v1, v2);
    std::cout << "Cross Product: ";
    for (int value : cross) {
        std::cout << value << " ";
    }
    std::cout << std::endl;

    return 0;
}